<a href="https://colab.research.google.com/github/naga-sk/DataScienceTraining/blob/main/Pandas_Excercise.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Import the pandas library  
Read the data company.csv in pandas

In [44]:
import pandas as pd
df = pd.read_csv("company.csv")

Find the dimension of given data

In [27]:
df.shape

(149, 5)

List down all the columns in data frame

In [17]:
df.columns

Index(['Company', 'Age', 'Salary', 'Place', 'Country'], dtype='object')

List the top 10 rows in datframe

In [14]:
df.head(10)

Company   Age  Salary     Place Country
0      TCS  20.0     NaN   Chennai   India
1  Infosys  30.0     NaN    Mumbai   India
2      TCS  35.0  2300.0  Calcutta   India
3  Infosys  40.0  3000.0     Delhi   India
4      TCS  23.0  4000.0    Mumbai   India
5  Infosys   NaN  5000.0  Calcutta   India
6      TCS   NaN  6000.0   Chennai   India
7  Infosys  23.0  7000.0    Mumbai   India
8      TCS  34.0  8000.0  Calcutta   India
9      CTS  45.0  9000.0     Delhi   India

List the last 15 rows in datframe

In [22]:
df.tail(15)

Company   Age  Salary      Place Country
134  Tata Consultancy Services  31.0  8345.0     Mumbai   India
135                        CTS  40.0  9284.0   Calcutta   India
136                        NaN   NaN     NaN      Delhi   India
137                        CTS  31.0  2034.0     Mumbai   India
138                        CTS  44.0  3033.0     Cochin   India
139                 Congnizant  22.0  2934.0      Noida   India
140                    Infosys  44.0  4034.0  Hyderabad   India
141                        TCS  33.0  5034.0   Calcutta   India
142            Infosys Pvt Lmt  22.0  8202.0     Mumbai   India
143                        TCS  33.0  9024.0   Calcutta   India
144                    Infosys  22.0  8787.0   Calcutta   India
145                    Infosys  44.0  4034.0      Delhi   India
146                        TCS  33.0  5034.0     Mumbai   India
147                    Infosys  22.0  8202.0     Cochin   India
148                        NaN   NaN     NaN        NaN     NaN

Find the number of rows in dataframe

In [23]:
len(df)

149

Check the information of dataframe

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 149 entries, 0 to 148
Data columns (total 5 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   Company  140 non-null    object 
 1   Age      130 non-null    float64
 2   Salary   124 non-null    float64
 3   Place    134 non-null    object 
 4   Country  148 non-null    object 
dtypes: float64(2), object(3)
memory usage: 5.9+ KB


Check the basic Statistics of Dataframe. Give your inference from the stats

In [31]:
df.describe(include='all')

Company         Age       Salary   Place Country
count      140  130.000000   124.000000     134     148
unique       6         NaN          NaN      11       1
top        TCS         NaN          NaN  Mumbai   India
freq        53         NaN          NaN      37     148
mean       NaN   30.484615  5312.467742     NaN     NaN
std        NaN   11.096640  2573.764683     NaN     NaN
min        NaN    0.000000  1089.000000     NaN     NaN
25%        NaN   22.000000  3030.000000     NaN     NaN
50%        NaN   32.500000  5000.000000     NaN     NaN
75%        NaN   37.750000  8000.000000     NaN     NaN
max        NaN   54.000000  9876.000000     NaN     NaN

In [30]:
df.agg

<bound method DataFrame.aggregate of      Company   Age  Salary     Place Country
0        TCS  20.0     NaN   Chennai   India
1    Infosys  30.0     NaN    Mumbai   India
2        TCS  35.0  2300.0  Calcutta   India
3    Infosys  40.0  3000.0     Delhi   India
4        TCS  23.0  4000.0    Mumbai   India
..       ...   ...     ...       ...     ...
144  Infosys  22.0  8787.0  Calcutta   India
145  Infosys  44.0  4034.0     Delhi   India
146      TCS  33.0  5034.0    Mumbai   India
147  Infosys  22.0  8202.0    Cochin   India
148      NaN   NaN     NaN       NaN     NaN

[149 rows x 5 columns]>

Retrieve 20 to 100 rows and Company and age usig iloc

In [34]:
df[['Company','Age']].iloc[20:100]

Company   Age
20      TCS  37.0
21  Infosys  50.0
22      TCS  21.0
23  Infosys   NaN
24      TCS   NaN
..      ...   ...
95  Infosys   NaN
96      TCS   NaN
97  Infosys  22.0
98      TCS  33.0
99      CTS  44.0

[80 rows x 2 columns]

Retrieve 20 to 100 rows and Company and age usig loc

In [36]:
df[['Company','Age']].loc[20:99]

Company   Age
20      TCS  37.0
21  Infosys  50.0
22      TCS  21.0
23  Infosys   NaN
24      TCS   NaN
..      ...   ...
95  Infosys   NaN
96      TCS   NaN
97  Infosys  22.0
98      TCS  33.0
99      CTS  44.0

[80 rows x 2 columns]

Chage the data in 100 row and Place as 'Noida'

In [45]:
df.loc[100]

Company       CTS
Age          22.0
Salary        NaN
Place      Cochin
Country     India
Name: 100, dtype: object

In [46]:
df.at[100,'Place'] = 'Noida'

In [47]:
df.loc[100]

Company      CTS
Age         22.0
Salary       NaN
Place      Noida
Country    India
Name: 100, dtype: object

Change the column 'Place' to 'City'

In [48]:
df.columns

Index(['Company', 'Age', 'Salary', 'Place', 'Country'], dtype='object')

In [49]:
df.rename(columns = {'Place':'City'}, inplace = True)

In [50]:
df.columns

Index(['Company', 'Age', 'Salary', 'City', 'Country'], dtype='object')

List down the unique data in each columns and find length of unique data

In [55]:
sr = pd.Series(df.squeeze(), name="company")
sr
sr.unique()

ValueError: ignored

Rename all the possible labels of column in Company as three labels
TCS
CTS
Infosys

In [ ]:
#Where ever you see age as 0 replace with NA

##Hint df[df.Age==20] = np.isnan

Check how many duplicated data is there?

[link text](https://)Remove all duplicated row-wise data

Remove the column 'Country'

Remove the row number 137

Find number of each labels in Company

Find number of each labels in City

Find Number of Null Values in each column

Remove all Null values in Salary

Replace the Null values in Comapany with mode

Replace the null value in Salary with median

Replace the null value in age with mean

Filter the data with age>40 and Salary<5000

Draw the pie chart for place and Company

Find the mean age with repective to the Company

Draw an histogram chart for age, Salary

Find the mean, count, median of salary with repective to the place